src/app/TideIndexApplication.cpp

```diff
@@ -1,1183 +1,1207 @@
+      carp(CARP_INFO, "Processed %d protein sequences", curProtein+1);
+    }
+  }
+  if (sort_on_disk)
+    fclose(fp);
+
+  if (targetsGenerated == 0) {
+    carp(CARP_FATAL, "No target sequences generated.  Is \'%s\' a FASTA file?",
+         fasta.c_str());
+  }
+  if (invalidPepCnt > 0) {
+    carp(CARP_INFO, "Ignoring %u peptide sequences containing unrecognized characters.", invalidPepCnt);
+  }
+  carp(CARP_INFO, "Generated %u targets, including duplicates.", targetsGenerated);
+
+  // Sort peptides
+  carp(CARP_INFO, "Sorting peptides");
+  
+  if (sort_on_disk) {
+    
+    remove(pathSortedPeptideFile.c_str());
+    #ifdef _WIN32
+      std::cout << "Windows OS\n";
+	  std::string import_csv = "Import-Csv -Header Mass, Protein, A, B, C -Delimiter ',' " + std::string(pathPeptideFile);
+	  std::string actual_sorting = "Select-Object { [int]$_.Mass }, Protein, A,B,C";
+	  std::string write_to_file = "ConvertTo-Csv -NoTypeInformation | Select-Object -Skip 1 | Set-Content -Path " + std::string(pathSortedPeptideFile);
+	  std::string cmd = "Powershell -Command \"" + import_csv + "|" + actual_sorting + "|" + write_to_file + "\"";
+	  int systemResult = system(cmd.c_str());
+	  if (systemResult != 0)
+		  carp(CARP_FATAL, "System sort failed, the call returned code: %i", systemResult);
+    #elif __linux__
+	  std::cout << "Linux Based OS\n";
+      std::string cmd = "sort  -t, -k 1n,1 -k 2,2 -o " +  std::string(pathSortedPeptideFile) + " " + std::string(pathPeptideFile);
+      int systemResult = system(cmd.c_str()); 
+      if (systemResult != 0) 
+        carp(CARP_FATAL, "System sort failed, the call returned code: %i", systemResult);
+    #elif __unix__
+      std::cout << "Other UNIX OS\n";
+	  std::string cmd = "sort  -t, -k 1n,1 -k 2,2 -o " + std::string(pathSortedPeptideFile) + " " + std::string(pathPeptideFile);
+	  int systemResult = system(cmd.c_str());
+	  if (systemResult != 0)
+		  carp(CARP_FATAL, "System sort failed, the call returned code: %i", systemResult);
+    #elif __APPLE__
+      std::cout << "Apple OS\n";
+	  std::string cmd = "sort  -t, -k 1n,1 -k 2,2 -o " + std::string(pathSortedPeptideFile) + " " + std::string(pathPeptideFile);
+	  int systemResult = system(cmd.c_str());
+	  if (systemResult != 0)
+		  carp(CARP_FATAL, "System sort failed, the call returned code: %i", systemResult);
+    #else
+      std::cout << "OS not supported";
+	  carp(CARP_FATAL, "OS not supported");
+	  std::string cmd = ""
+    #endif
+		 
+    bool is_exist = boost::filesystem::exists(pathSortedPeptideFile);
+	  if (is_exist) {
+		  uintmax_t sorted_peptide_file_size = boost::filesystem::file_size(pathSortedPeptideFile);
+		  uintmax_t unsorted_peptide_file_size = boost::filesystem::file_size(pathSortedPeptideFile);
+		  if (sorted_peptide_file_size != unsorted_peptide_file_size) {
+			  carp(CARP_FATAL, "Something went wrong whilst sorting, because sorted peptide file and peptide files have different sizes");

```

Comment:
Something went wrong while sorting: original peptide file and sorted peptide file have different lengths.
Original contains XXX peptides, but sorted contains XXX peptides.

Reply:
It may require extra processing to find out the number of peptides in these files, hence I changed it to:

Something went wrong while sorting: original peptide file and sorted peptide file have different lengths.
Original file size is XXX, but sorted file size is XXX.